tests/mypy_test.py

```diff
@@ -0,0 +1,89 @@
+import glob
+import os
+import re
+from subprocess import Popen, PIPE
+from typing import Text, List, Iterable, Tuple, Any
+
+import pytest
+import yaml
+
+
+DEFAULT_PATH = 'main.py'
+ENCODING = 'utf-8'
+
+
+def load_test_cases(root: Text) -> Iterable[Tuple[Text, Text, List[dict]]]:
+    paths = glob.glob(os.path.join(root, '*.yml'))
+    assert paths, 'No test data found in %s' % os.path.abspath(root)
+
+    for path in paths:
+        with open(path, 'rb') as fd:
+            raw_data = yaml.load(fd)
+        for case in raw_data:
+            _, filename = os.path.split(path)
+            name = case.get('case')
+            assert name, 'Unnamed case: %r' % case
+            files = case.get('files', [])
+            unique = {file.get('path', DEFAULT_PATH) for file in files}
+            assert len(files) == len(unique), 'Duplicate files in %s' % name
+            yield filename, name, files
+
+
+def inline_errors(errors: List[Text], files: List[dict]) -> List[dict]:
+    contents = {
+        file.get('path', DEFAULT_PATH): strip_errors(file.get('content', ''))
+        for file in files}
+
+    for error in errors:
+        index, message, path = parse_error_line(error)
+        content = contents.get(path)
+        assert content, 'Cannot find contents of %s' % path
+        lines = content.splitlines(keepends=True)
+        assert 0 <= index < len(lines)
+        line = lines[index]
+        updated = '%s# E: %s\n' % (line.rstrip('\n'), message)
+        lines[index] = updated
+        contents[path] = ''.join(lines)
+
+    results = [file.copy() for file in files]
+    for result in results:
+        result['content'] = contents.get(result.get('path', DEFAULT_PATH), '')
+    return results
+
+
+def strip_errors(s: Text) -> Text:
+    return re.sub(r'#\s*E:.*', r'', s)
+
+
+def parse_error_line(line: Text) -> Tuple[int, Text, Text]:
+    m = re.match(r'(.+):([0-9]+): (error): (.+)', line)
+    assert m, 'Cannot parse output line: %s' % line
+    path = m.group(1)
+    index = int(m.group(2)) - 1
+    message = m.group(4)
+    return index, message, path
+
+
+@pytest.mark.parametrize('filename,case,files', load_test_cases('test-data'))
```

Comment:
This appears to be calling load_test_cases() even if this test were not selected. Upon further inspection load_test_cases() is a generator function so the call doesn't actually do any work, but it still creeped me out. (Maybe load_test_cases() deserves a comment explaining why it *should* be a generator.)